tests/noto_fonts_for_android_test.py

```diff
@@ -158,4 +165,21 @@ def test_font_full_weight_coverage():
 
 
 def test_font_psnames():
-  pass
+  root = etree.parse(str(_noto_4_android_file()))
+  errors = []
+  font_to_xml_psnames = collections.defaultdict(set)
+
+  for font_el in root.xpath("//font[@path]"): 
+    path = _font_path(font_el)
+    psname = _psname(font_el)
+    font_to_xml_psnames[(path, font_el.attrib.get("index", -1))].add(str(psname))
+
+  for (font_path, font_number), xml_psname in font_to_xml_psnames.items():
+    font = _open_font_path(font_path, font_number)
+    postscript_names = get_name_entry_strings(font, _POSTSCRIPT_NAME)
+    assert len(postscript_names) > 0
+    for el in xml_psname:
+      if not (el in postscript_names):
+        errors.append(f"postScriptName {postscript_names} in {font_path} doesn't match the entry in XML {el}")
+
```

Comment:
since each <font> can be validated in isolation I think I'd expect you can loop over them once? - ignoring VFs momentarily I think you end up with something like this:

```
for font_el in ...
  xml_psname = _psname
  path = _font_path
  font = _open_font_path(_font_path(font_el), ...get index...)
  ps_names = set(get_name_entry_string))
  if len(ps_names) != 1
    # record an error, we should have exactly 1 psname
  elif {xml_psname} != ps_names:
    # record an error, wroing psname
```

Reply:
Just realized they want the same psname for all vf fonts so maybe the above is pretty much OK

Reply:
Changed to record an error when len(ps_names) != 1